hs_composite_resource/models.py

```diff
@@ -620,6 +1022,12 @@ def update_temporal_coverage(self):
 
         update_target_temporal_coverage(self)
 
+    def get_model_program_aggregations(self):
+        """Gets a list of model program aggregations in this (self) resource"""
+
+        mp_aggregations = [aggr for aggr in self.logical_files if aggr.type_name() == "ModelProgramLogicalFile"]
```

Comment:
i get that we're already doing things like this for the other types, but we need to be careful about performance.  Anytime we can filter directly in a query instead of loading into python objects, we should be doing the filter in the query.